app/src/main/java/com/github/se/gatherspot/ui/CreateEvent.kt

```diff
@@ -0,0 +1,265 @@
+                      label = { Text("Min Attendees") },
+                      placeholder = { Text("Min Attendees") })
+                  // Max attendees
+                  OutlinedTextField(
+                      modifier =
+                          Modifier.width(WIDTH_2ELEM).height(HEIGHT).testTag("inputMaxAttendees"),
+                      value = maxAttendees,
+                      onValueChange = { maxAttendees = it },
+                      label = { Text("Max Attendees") },
+                      placeholder = { Text("Max Attendees") })
+                }
+
+            // Inscription limit date
+            OutlinedTextField(
+                modifier =
+                    Modifier.width(WIDTH).height(HEIGHT).testTag("inputInscriptionLimitDate"),
+                value = inscriptionLimitDate,
+                onValueChange = { inscriptionLimitDate = it },
+                label = { Text("Inscription Limit Date") },
+                placeholder = { Text(EventFirebaseConnection.DATE_FORMAT) })
+            // Inscription limit time
+            OutlinedTextField(
+                modifier =
+                    Modifier.width(WIDTH).height(HEIGHT).testTag("inputInscriptionLimitTime"),
+                value = inscriptionLimitTime,
+                onValueChange = { inscriptionLimitTime = it },
+                label = { Text("Inscription Limit Time") },
+                placeholder = { Text(EventFirebaseConnection.TIME_FORMAT) })
+
+            // TODO :Upload images
+
+            // Button to create the event
+            Button(
+                onClick = {
+                  try {
+                    eventViewModel.validateAndCreateEvent(
+                        title.text,
+                        description.text,
+                        location!!,
+                        eventStartDate.text,
+                        eventEndDate.text,
+                        eventTimeStart.text,
+                        eventTimeEnd.text,
+                        maxAttendees.text,
+                        minAttendees.text,
+                        inscriptionLimitDate.text,
+                        inscriptionLimitTime.text)
+                  } catch (e: Exception) {
+                    // Display error message
+                    errorMessage = e.message.toString()
+                    showErrorDialog = true
+                  }
+                },
+                modifier = Modifier.width(WIDTH).height(HEIGHT).testTag("createEventButton"),
+                enabled =
+                    (title.text != "") &&
+                        (description.text != "") &&
+                        (eventStartDate.text != "") &&
+                        (eventTimeStart.text != "") &&
+                        (eventTimeEnd.text != ""),
+                shape = RoundedCornerShape(size = 10.dp)) {
+                  Text(text = "Create event")
+                }
+          }
+        }
+        if (showErrorDialog) {
```

Comment:
Playing with a general Exception makes the job but pay attention here that the code catch any exception, and that can be problematic as it could hides a network one for example.

Just implementing a more specific and custom exception to be catched here, will avoid that problem.

Reply:
If there is a network error, this will also be displayed in the alertDialog composable , what different behaviour should we adopt in case a network error appears.